azure/microservice-ecommerce/xebialabs/xld-terraform-apps.yaml.tmpl

```diff
@@ -0,0 +1,66 @@
+{{$app := .AppName | kebabcase}}
+{{$clusterName := .ClusterName | kebabcase}}
+apiVersion: xl-deploy/v1
+kind: Applications
+spec:
+- name: {{$app}}
+  type: core.Directory
+  children:
+{{if .ProvisionCluster }}
+# Provision AKS cluster using Terraform templates
+  - name: AKS-TERRAFORM
+    type: core.Directory
+    children:
+    - name: {{$app}}-terraform
+      type: udm.Application
+      children:
+      - name: '1.0.0'
+        type: udm.DeploymentPackage
+        deployables:
+        - name: {{$app}}
+          type: terraform.Module
+          file: !file ../terraform
+          inputVariables:
+            name: {{$app}}
+            resource_group_name: {{.ResourceGroup}}
+            cluster_name: {{.ClusterName}}
+            location: {{.AKSRegion}}
+            client_id: {{.ClientID}}
+            client_secret: !value ClientSecret
+            subscription_id: {{.SubscriptionID}}
+            tenant_id: {{.TenantID}}
+          boundTemplates:
+          - "../{{$clusterName}}"
+          - "../{{$clusterName}}-env"
+
+        templates:
+        # creates the required XLD infrastructure entries
+        - name: {{$clusterName}}
```

Comment:
Please append/prepend something to the clusterName so that if we use same appname/clustername this will not throw an error as XLD can't have duplicate CI names, see the GCP blueprint

Reply:
`{{$clusterName}}-master`